runtime/kusama/src/lib.rs

```diff
@@ -620,13 +622,32 @@ impl pallet_staking::Config for Runtime {
 	type NextNewSession = Session;
 	type MaxNominatorRewardedPerValidator = MaxNominatorRewardedPerValidator;
 	type OffendingValidatorsThreshold = OffendingValidatorsThreshold;
-	type VoterList = VoterList;
+	type VoterList = VoterBagsList;
+	type TargetList = UseValidatorsMap<Self>;
 	type MaxUnlockingChunks = frame_support::traits::ConstU32<32>;
 	type BenchmarkingConfig = runtime_common::StakingBenchmarkingConfig;
 	type OnStakerSlash = NominationPools;
 	type WeightInfo = weights::pallet_staking::WeightInfo<Runtime>;
 }
 
+const OLD_PALLET_NAME:  &'static str = "VoterList";
+pub struct StakingMigrationV11;
```

Comment:
Would make the boilerplate less if you move the struct that implements `OnRuntimeUpgrade` to substrate and only use it here.

Reply:
yeah, but I need to pass the pallet name. Which is not possible within the struct due to the absence of self reference in those methods.